support-workers/src/test/scala/com/gu/zuora/BuildDigitalPackSubscriptionSpec.scala

```diff
@@ -0,0 +1,102 @@
+import com.gu.zuora.ProductSubscriptionBuilders._
+import org.joda.time.LocalDate
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatestplus.mockito.MockitoSugar._
+
+import scala.concurrent.Future
+
+class BuildDigitalPackSubscriptionSpec extends AsyncFlatSpec with Matchers {
+
+  "SubscriptionData for a corporate subscription" should "be correct" in
+    corporate.map { subData =>
+      subData shouldBe SubscriptionData(
+        List(RatePlanData(RatePlan("2c92c0f971c65dfe0171c6c1f86e603c"), List(), List())),
+        Subscription(
+          contractAcceptanceDate = saleDate,
+          contractEffectiveDate = saleDate,
+          termStartDate = saleDate,
+          createdRequestId__c = "f7651338-5d94-4f57-85fd-262030de9ad5",
+          autoRenew = true,
+          initialTermPeriodType = Month,
+          initialTerm = 12,
+          renewalTerm = 12,
+          termType = "TERMED",
+          readerType = ReaderType.Corporate,
+          promoCode = None,
+          redemptionCode = Some("CODE"),
+          corporateAccountId = Some("1")
+        )
+      )
+    }
+
+  "SubscriptionData for a monthly subscription" should "be correct" in
+    monthly.map { subData =>
+      subData shouldBe SubscriptionData(
+        List(RatePlanData(RatePlan("2c92c0f84bbfec8b014bc655f4852d9d"), List(), List())),
+        Subscription(
+          contractAcceptanceDate = saleDate.plusDays(16),
+          contractEffectiveDate = saleDate,
+          termStartDate = saleDate,
+          createdRequestId__c = "f7651338-5d94-4f57-85fd-262030de9ad5",
+          autoRenew = true,
+          initialTermPeriodType = Month,
+          initialTerm = 12,
+          renewalTerm = 12,
+          termType = "TERMED",
+          readerType = ReaderType.Direct,
+          promoCode = None,
+          redemptionCode = None,
+          corporateAccountId = None
+        )
+      )
+    }
+
+  lazy val promotionService = mock[PromotionService]
+  lazy val saleDate = new LocalDate(2020, 6, 5)
+
+  lazy val corporate = buildDigitalPackSubscription(
+    DigitalPack(GBP, null /* FIXME should be Option-al for a corp sub */ , Corporate),
+    UUID.fromString("f7651338-5d94-4f57-85fd-262030de9ad5"),
+    Country.UK,
+    Right(CorporateRedemption("CODE", "1")),
+    promotionService,
+    DEV,
+    isTestUser = false,
+    new GetCodeStatus({
+      case "CODE" => Future.successful(Some(Map(
+        "available" -> DynamoLookup.DynamoBoolean(true),
+        "corporateId" -> DynamoLookup.DynamoString("1")
+      )))
+    }),
+    () => saleDate
+  ).value.map(_.right.get)
+
+  lazy val monthly = buildDigitalPackSubscription(
+    DigitalPack(GBP, Monthly),
+    UUID.fromString("f7651338-5d94-4f57-85fd-262030de9ad5"),
+    Country.UK,
+    Left((ZuoraDigitalPackConfig(14, 2), None)),
+    promotionService,
+    DEV,
+    isTestUser = false,
+    new GetCodeStatus(_ => Future.failed(new Throwable)),
```

Comment:
technically the GetCodeStatus is only relevant to corporate subs, so would ideally only be included somehow in the `Right` rather than having to be also specified for `Left` standard subs.
But I'd rather keep the side effects separate, and I'd also rather not do anything too fancy with the types to keep them separate but checked together.  Would certainly end up involving implicits.
Other ideas welcome though.